File: lista_1/ex9.py
n = int(input("Insira um número para descobrir o seu fatorial: "))

fat = 0

i = 1

res = n

if n == 0:
    fat = 0
else:
    for i in range(i, n):
        fat = (n - 1)
        res *= fat


print(f"o fatorial de {n} é: {res}")

File: C_parte2/lista_4/ex9.c
#include <stdio.h>


int main(void)
{
    int i, n, fat, total = 0;

    printf("Insira um número para descobrir seu fatorial: ");
    scanf("%d", &n);

    fat = n;

    if (n == 0 || n == 1)
    {
        fat = 1;
    }

    for (i = n; i > 0; i--)
    {
        total = fat *= i;
    }

    printf("Fatorial de %d é: %d\n", n, total);

    return 0;
}
